### PR TITLE
Feature: Split PR & issues into separate events

### PR DESCRIPTION
**Pull request by BunnyTheLifeguard**
- [x] read and understood the [Contributing Guidelines][contributing]
- [x] Updated any nessecary files such as the README.md and/or CHANGELOG.md.

## Type of Pull request

- [ ] **Minor Change**  
  This Pull request doesn't break existing configuration.
- [ ] **Major Change**  
  This Pull request will break existing configuration.
- [ ] **Bug fix**  
  This Pull request will fix a (critical) bug.
- [x] **Documentation**  
  This Pull request only changes documentation (README.md, CHANGELOG.md, etc.)
- [ ] **Other:** `__________` 

## Description

This PR splits the pr and issue events into different events so users can better include/exclude specific events.

## To Do:

- Update changelog after Merge/Release

[contributing]: https://github.com/Readme-Workflows/.github/blob/main/.github/CONTRIBUTING.md

### CODE REVIEW

**Comment by abhijoshi2k** [comment]
For backward compatibility, we shouldn't discard the original pr and issue events.
If the older events are excluded, they should block all sub events.

For example, if pr is excluded, it should block pr_opened, pr_merged and pr_closed events.

**Comment by BunnyTheLifeguard** [comment]
> For backward compatibility, we shouldn't discard the original pr and issue events. If the older events are excluded, they should block all sub events.
> 
> For example, if pr is excluded, it should block pr_opened, pr_merged and pr_closed events.

Makes sense. I will look into it.
Btw. how does one test the workflow? Tried with act but always run into this error:

```bash
act -s GITHUB_TOKEN=ghp_xyz... -j build -W .github/workflows/test-workflow.yml

docker exec cmd=[node /var/run/act/actions/Readme-Workflows-recent-activity@main/dist/index.js] user= workdir=
| serializers
| {
|   PushEvent: [Function: PushEvent],
|   IssuesEvent: [Function: IssuesEvent],
|   PullRequestEvent: [Function: PullRequestEvent],
|   CreateEvent: [Function: CreateEvent],
|   ForkEvent: [Function: ForkEvent],
|   GollumEvent: [Function: GollumEvent],
|   MemberEvent: [Function: MemberEvent],
|   PullRequestReviewEvent: [Function: PullRequestReviewEvent],
|   ReleaseEvent: [Function: ReleaseEvent],
|   WatchEvent: [Function: WatchEvent]
| }
| ✖  error     Event `push` is not supported by this action.
```

**Comment by abhijoshi2k** [comment]
You can run the workflow on your own repo to test it.
For example, you can run the workflow by editing your workflow file as: `BunnyTheLifeguard/recent-activity@feat/split-pr-events`

File: src/serializers.js [hunk 1]
```diff
@@ -10,8 +10,11 @@ const IssueCommentEvent = require("./events/IssueCommentEvent");
 const CommitCommentEvent = require("./events/CommitCommentEvent");
 const PullRequestReviewCommentEvent = require("./events/PullRequestReviewCommentEvent");
 const PushEvent = require("./events/PushEvent");
-const IssuesEvent = require("./events/IssuesEvent");
-const PullRequestEvent = require("./events/PullRequestEvent");
+const IssuesOpenEvent = require("./events/IssuesOpenEvent");
```
Comment (PuneetGopinath) [inline src/serializers.js:13]: This is where the problem lies. GitHub API does not provide a type of the value "IssuesOpenEvent" or the other new names that you have made. Due to this change, while testing I see that none of the issue and pull request-related events are shown in the profile.
You can checkout https://docs.github.com/en/developers/webhooks-and-events/github-event-types for the docs

Comment (PuneetGopinath) [reply src/serializers.js:13]: You have to modify the functions/filterContent.js file to use the respective serializers for their respective events

Comment (PuneetGopinath) [reply src/serializers.js:13]: You don't need to close it, let it be open. Whenever free you can look into it, that's not an problem.
We appreciate your efforts to contribute to Readme-Workflows!

**Comment by BunnyTheLifeguard** [comment]
Reverted the Events acccording to the GH API and implemented a filter to extract disabled issues/pr actions.

**Comment by PuneetGopinath** [comment]
> Reverted the Events according to the GH API and implemented a filter to extract disabled issues/pr actions.

Sorry, I read the comment and the forgot it totally.
Did you test it? If so can you show the workflow results

I'm really very sorry for being inactive.

**Comment by abhijoshi2k** [comment]
@PuneetGopinath I think we can change that to 2023. Feel free to do the change before release.

**Comment by PuneetGopinath** [comment]
@Andre601 Can you do the release? I don't remember anything after a year of not touching github 😄

**Comment by Andre601** [comment]
> @Andre601 Can you do the release? I don't remember anything after a year of not touching github 😄

Me neither. Need to look at this.
What release version should it be?

**Pull request merged**